### PR TITLE
[Java] Stop storing the config object and add a migration to delete its old file

### PR DESCRIPTION
**Pull request by arifBurakDemiray**

### CODE REVIEW

**Comment by arifBurakDemiray** [comment]
[[Java] Stop storing the config object and add a migration to delete its old file](https://trello.com/c/4T9jcdTY/847-java-stop-storing-the-config-object-and-add-a-migration-to-delete-its-old-file)

File: sdk-java/src/main/java/ly/count/sdk/java/internal/MigrationHelper.java [hunk 1]
```diff
@@ -71,13 +76,96 @@ protected void applyMigrations(final Map<String, Object> migrationParams) {
...
+                    idStream.readInt(); // realm
+                    DeviceIdType type = DeviceIdType.fromInt(idStream.readInt());
+                    Object deviceId = idStream.readObject();
+                    if (type != null && deviceId instanceof String && !Utils.isEmptyOrNull((String) deviceId)) {
+                        storageProvider.setDeviceIdType(type.name());
+                        storageProvider.setDeviceID((String) deviceId);
+                    }
+                }
+            }
+            Files.delete(configFile.toPath());
```
Comment (ArtursKadikis) [inline sdk-java/src/main/java/ly/count/sdk/java/internal/MigrationHelper.java:118]: we want to delete the file even if there was a exception

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,8 @@
   * 'enableRemoteConfigAutomaticTriggers' to automatically download remote config values on init
   * 'remoteConfigRegisterGlobalCallback(RCDownloadCallback callback)' to register a remote config callback
 * Added the ability to set the user profile picture with an URL
+* Added to a way to get device id type by calling "Countly.instance().getDeviceIdType()" call.
+* Config storage is deleted, only device id and device id type are stored now
```
Comment (ArtursKadikis) [inline CHANGELOG.md:18]: b etter entry

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
There should be test cases for both device ID type

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
Missing a test case for when a custom device Id is being provided

**Pull request merged**